standard/expressions.md

```diff
@@ -2064,9 +2122,9 @@ A *this_access* is permitted only in the *block* of an instance constructor, an
 - When `this` is used in a *primary_expression* within an instance method or instance accessor of a class, it is classified as a value. The type of the value is the instance type ([§15.3.2](classes.md#1532-the-instance-type)) of the class within which the usage occurs, and the value is a reference to the object for which the method or accessor was invoked.
 - When `this` is used in a *primary_expression* within an instance constructor of a struct, it is classified as a variable. The type of the variable is the instance type ([§15.3.2](classes.md#1532-the-instance-type)) of the struct within which the usage occurs, and the variable represents the struct being constructed.
   - If the constructor declaration has no constructor initializer, the `this` variable behaves exactly the same as an `out` parameter of the struct type. In particular, this means that the variable shall be definitely assigned in every execution path of the instance constructor.
-  - Otherwise, the `this` variable behaves exactly the same as a `ref` parameter of the struct type. In particular, this means that the variable is considered initially assigned.
+  - Otherwise, the `this` variable behaves exactly the same as a `in` parameter of the struct type. In particular, this means that the variable is considered initially assigned, and can't be reassigned.
```

Comment:
Unless this changes in C# 8+, this doesn't appear to be the case:

```csharp
struct Test
{
    Test(int t) : this()
    {
        // Constructor with a constructor initializer, but still able to reassign this afterwards
        this = new Test("h");
    }

    Test(string x)
    {
    }
}
```

Reply:
I think you're right. Let's discuss, but I think this change should be reverted.

Reply:
Action 2023-04-19: this needs to be for members of readonly structs, other than the constructor.
@BillWagner to fix.